File: SDKs/googleapis.com/drive/v3/go/pkg/models/operations/drivedrivesunhide.go
package operations

import (
"openapi/pkg/models/shared")

type DriveDrivesUnhidePathParams struct {
    DriveID string `pathParam:"style=simple,explode=false,name=driveId"`
    
}

type DriveDrivesUnhideQueryParams struct {
    Alt *shared.AltEnum `queryParam:"style=form,explode=true,name=alt"`
    Fields *string `queryParam:"style=form,explode=true,name=fields"`
    Key *string `queryParam:"style=form,explode=true,name=key"`
    OauthToken *string `queryParam:"style=form,explode=true,name=oauth_token"`
    PrettyPrint *bool `queryParam:"style=form,explode=true,name=prettyPrint"`
    QuotaUser *string `queryParam:"style=form,explode=true,name=quotaUser"`
    UserIP *string `queryParam:"style=form,explode=true,name=userIp"`
    
}

type DriveDrivesUnhideSecurity struct {
    Oauth2 shared.SchemeOauth2 `security:"scheme,type=oauth2"`
    Oauth2c shared.SchemeOauth2c `security:"scheme,type=oauth2"`
    
}

type DriveDrivesUnhideRequest struct {
    PathParams DriveDrivesUnhidePathParams 
    QueryParams DriveDrivesUnhideQueryParams 
    Security DriveDrivesUnhideSecurity 
    
}

type DriveDrivesUnhideResponse struct {
    ContentType string 
    Drive *shared.Drive 
    StatusCode int64 
    
}

File: SDKs/googleapis.com/drive/v2/typescript/src/sdk/models/operations/drivedrivesunhide.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class DriveDrivesUnhidePathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=driveId" })
  driveId: string;
}


export class DriveDrivesUnhideQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=alt" })
  alt?: shared.AltEnum;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=fields" })
  fields?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=key" })
  key?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=oauth_token" })
  oauthToken?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=prettyPrint" })
  prettyPrint?: boolean;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=quotaUser" })
  quotaUser?: string;

  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=true;name=userIp" })
  userIp?: string;
}


export class DriveDrivesUnhideSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2: shared.SchemeOauth2;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  oauth2c: shared.SchemeOauth2c;
}


export class DriveDrivesUnhideRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: DriveDrivesUnhidePathParams;

  @SpeakeasyMetadata()
  queryParams: DriveDrivesUnhideQueryParams;

  @SpeakeasyMetadata()
  security: DriveDrivesUnhideSecurity;
}


export class DriveDrivesUnhideResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  drive?: shared.Drive;

  @SpeakeasyMetadata()
  statusCode: number;
}
